Upgrade to latest rom-4

GustavoCaso opened the pull request.

GustavoCaso commented:
Removing the `Transproc` part makes the fail by not returning the keys as symbols, but do we need actually always return the keys as symbols?

cflipse commented on lib/rom/yaml/schema.rb:
```diff
@@ -0,0 +1,19 @@
+
+module ROM
+  module YAML
+    # Elasticsearch relation schema
```
Not elasticsearch

cflipse commented:
I think we do -- it's a matter of consistency between relations, and having to know that _this_ relation is of _this_ type, and therefore has _these_ keys ... that's way too much cognitive overhead, that has to go into _way_ too many locations -- consider the arguments to joining relation data in the `combine` commands ...

Commit:
Depend on rom-core ~> 4.1

4.1 includes a fix for multi-adapter setup so it's better to enforce it

Commit:
Update .rspec

Commit:
Update adapter spec

Commit:
Upgrade to latest rom-4

solnic commented:
I restored symbolizing keys. It's because using query API would require using string keys, which would be really weird. In the future we should improve memory adapter (which this one is based on), so that it uses attribute API in query methods.

Commit:
Remove enforced output_schema

It turned out we can't use it just yet, because memory adapter doesn't
really make use of attribute API from schemas. This means that it
doesn't translate key names correctly based on schema definition.

wafcio commented on .rspec:
```diff
@@ -1,2 +1,3 @@
 --color
 --order random
+--require ./spec/spec_helper.rb
```
in this case it will be good to remove this line from spec files
